pyexasol/connection.py

```diff
@@ -266,25 +401,119 @@ def set_autocommit(self, val):
         })
 
     def set_query_timeout(self, val):
+        """
+        Set the maximum time in seconds for which a query can run before Exasol kills it automatically. 
+
+        Args:
+            val:
+                Timeout value in seconds.
+                Set value ``0`` to disable timeout.
+
+        Note:
+            It is highly recommended to set timeout for UDF scripts to
+            avoid potential infinite loops and very long transactions.
+        """
         self.set_attr({
             'queryTimeout': int(val)
         })
 
     def open_schema(self, schema):
+        """
+        Wrapper for `OPEN SCHEMA`
+
+        Args:
+            schema: Schema name
+        """
         self.set_attr({
             'currentSchema': self.format.default_format_ident_value(schema)
         })
 
     def current_schema(self):
+        """
+        Get the name of the current schema.
+
+        Returns:
+            Name of currently opened schema. Return empty string if no schema was opened.
+        """
         return self.attr.get('currentSchema', '')
 
     def export_to_file(self, dst, query_or_table, query_params=None, export_params=None):
+        """
+        Export large amount of data from Exasol to file or file-like object using fast HTTP transport.
+
+        Note:
+            File must be opened in binary mode.
+
+        Args:
+            dst:
+                Path to file or file-like object.
+            query_or_table:
+                SQL query or table for export.
+            query_params:
+                Values for SQL query placeholders.
+            export_params:
+                Custom parameters for Export query.
+
+        Examples:
+            >>> con = ExaConnection(...)
+            >>> with open('/tmp/file.csv', 'wb') as f:
+            ...     con.export_to_file(
+            ...         dst=f,
+            ...         query_or_table="SELECT * FROM table"
+            ...     )
+        """
         return self.export_to_callback(cb.export_to_file, dst, query_or_table, query_params, None, export_params)
 
     def export_to_list(self, query_or_table, query_params=None, export_params=None):
```

Comment:
Maybe add the return type hint?

Reply:
No hint can be given, depends on the type returned bye the callback as far as I understand.